src/sentry/migrations/0764_migrate_bad_status_substatus_rows.py

```diff
@@ -0,0 +1,170 @@
+# Generated by Django 5.1.1 on 2024-09-17 21:16
+
+from datetime import timedelta
+
+from django.apps.registry import Apps
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.utils import timezone
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBarApprox
+
+
+class ActivityType:
+    SET_IGNORED = 3
+
+
+class GroupHistoryStatus:
+    REGRESSED = 7
+    ARCHIVED_UNTIL_ESCALATING = 15
+    ARCHIVED_FOREVER = 16
+    ARCHIVED_UNTIL_CONDITION_MET = 17
+
+
+class GroupSubStatus:
+    # GroupStatus.IGNORED
+    UNTIL_ESCALATING = 1
+    # Group is ignored/archived for a count/user count/duration
+    UNTIL_CONDITION_MET = 4
+    # Group is ignored/archived forever
+    FOREVER = 5
+
+    # GroupStatus.UNRESOLVED
+    ESCALATING = 2
+    ONGOING = 3
+    REGRESSED = 6
+    NEW = 7
+
+
+class GroupStatus:
+    UNRESOLVED = 0
+    RESOLVED = 1
+    IGNORED = 2
+    PENDING_DELETION = 3
+    DELETION_IN_PROGRESS = 4
+    PENDING_MERGE = 5
+
+    # The group's events are being re-processed and after that the group will
+    # be deleted. In this state no new events shall be added to the group.
+    REPROCESSING = 6
+
+    # TODO(dcramer): remove in 9.0
+    MUTED = IGNORED
+
+
+UNRESOLVED_SUBSTATUS_CHOICES = {
+    GroupSubStatus.ONGOING,
+    GroupSubStatus.ESCALATING,
+    GroupSubStatus.REGRESSED,
+    GroupSubStatus.NEW,
+}
+
+IGNORED_SUBSTATUS_CHOICES = {
+    GroupSubStatus.UNTIL_ESCALATING,
+    GroupSubStatus.FOREVER,
+    GroupSubStatus.UNTIL_CONDITION_MET,
+}
+
+# End copy
+
+ACTIVITY_DATA_FIELDS = {
+    "ignoreCount",
+    "ignoreDuration",
+    "ignoreUntil",
+    "ignoreUserCount",
+    "ignoreUserWindow",
+    "ignoreWindow",
+}
+
+
+def fix_substatus_for__groups(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
```

Comment:
```suggestion
def fix_substatus_for_groups(apps: Apps, schema_editor: BaseDatabaseSchemaEditor) -> None:
```

Was the double `_` intentional?

Reply:
nope, good catch!